src/server/serverSetup.js

```diff
@@ -168,31 +205,53 @@ fluid.defaults("sjrk.storyTelling.server.app.storyTellingHandlers", {
             type: "sjrk.storyTelling.server.uploadsHandler",
             "route": "/*",
             "method": "get",
-            "prefix": "{server}.options.globalConfig.uploadedFilesHandlerPath"
+            "prefix": "{server}.options.secureConfig.uploadedFilesHandlerPath"
         },
         testsHandler: {
             type: "sjrk.storyTelling.server.testsHandler",
             "route": "/*",
             "prefix": "/tests",
             "method": "get"
         },
-        testBinariesHandler: {
-            type: "sjrk.storyTelling.server.testBinariesHandler",
-            "route": "/*",
-            "prefix": "/binaries",
-            "method": "get"
+        clientConfigHandler: {
+            type: "sjrk.storyTelling.server.clientConfigHandler",
+            route: "/clientConfig",
+            method: "get"
         },
-        uiHandler: {
-            type: "sjrk.storyTelling.server.uiHandler",
+        themeHandler: {
+            type: "sjrk.storyTelling.server.themeHandler",
             "route": "/*",
             "method": "get"
         }
     }
 });
 
-// Resolves a JSON file and parses it before
-// returning it
+/* Resolves a JSON file and parses it before returning it
+ * - "jsonFilePath": the path to the JSON file to parse
+ */
 sjrk.storyTelling.server.resolveJSONFile = function (jsonFilePath) {
     var file = kettle.resolvers.file(jsonFilePath);
     return JSON.parse(file);
 };
+
+/* Returns the path to the custom theme folder. The theme folder's name is
+ * expected to match the theme name being passed in. If the theme is not
+ * specified, the path returend will be the current directory ("."). If the
+ * theme is specified but the resolved folder doesn't exist within themeFolder,
+ * an error will be reported.
+ * - "theme": The name of the theme for which to find the path
+ * - "themeFolder": The folder/path that contains the theme being retrieved
+ */
+sjrk.storyTelling.server.getThemePath = function (theme, themeFolder) {
+    var themePath = ".";
+
+    if (theme) {
+        themePath = fluid.stringTemplate(themeFolder, { theme: theme });
+
+        if (!fs.existsSync(themePath)) {
+            fluid.fail("The custom theme folder " + themePath + " does not exist. Please verify that the theme name is configured properly.");
```

Comment:
The failure reported will include your message as well as the detailed tech error by printing out the component tree. The way I found to report a graceful error and exit is to replace this line with:

```
console.log("FATAL ERROR: ...");
process.exit(1);
```

Reply:
I implemented this suggestion, and while it did kill the server gracefully, it was not something I was able to test against to ensure that this was what was happening. Since the `process.exit(1)` call simply kills the server, I would have to use or write some sort of external tool to test the output in the console.

The detailed component tree printout is verbose and would make finding the actual error message difficult, but I think there's value in doing things the "Infusion way" and calling fluid.fail. I wonder if there is a way to call fluid.fail without having it print out the component tree, but just the message being passed into it.

@amb26 do you have any thoughts on this matter?